### PR TITLE
[4.0] [RFC] Sidebar links to navigation bar

### PR DESCRIPTION
**Pull request by todordev**
### Summary of Changes
This PR is another concept of the sidebar submenu ( #19552 ). It shows the links as navbar and adds a button that you will be able to use to show or hide that bar.

Before
35769494-71eb16d0-0914-11e8-9c95-643454370071

After
after

### CODE REVIEW

**Comment by dgrammatiko** [comment]
Can you please post a screenshot of the page I mentioned in your previous PR: https://github.com/joomla/joomla-cms/pull/19552#issuecomment-363217859

This is essential, as I think you're overlooking what the content of the sidebar might be!

**Comment by C-Lodder** [comment]
Much prefer this approach, although I dont think the toggle would be required at all. Also a small linear separator inbetween each item.

**Comment by ciar4n** [comment]
I like the idea. Only issues I can see with this approach..

1. On 3PD components which have a lot of menu items, this menu can potentially wrap to a 2nd line
2. @dgt41 point that this sidebar may contain more than menu items. Personally I would agree with https://github.com/joomla/joomla-cms/issues/16020#issuecomment-301502107 

An issue on the same subject.. https://github.com/joomla/joomla-cms/issues/16020

**Comment by Fedik** [comment]
This is looks good 😉

**Comment by todordev** [comment]
Here you are subnavigation on the view Languages Overrides. :)

languages_overrides

**Comment by dgrammatiko** [comment]
And as you see you're missing something there: 
screen shot 2018-02-07 at 23 00 45

Obviously I'm not against this (i like it, I think @ciar4n already had a mock up months ago) but this needs a decision if it's ok to break 3rd PD code and dictate to only use the new filters.
Good job here!

**Comment by brianteeman** [comment]
Just a note that the language overrides screen is the only one remaining that was not converted to use "search tools" See #11976

**Comment by todordev** [comment]
No one force the developers to replace the sidebar with "search tools". The sidebar still exists and everyone will be able to implement it in their extensions.

**Comment by dgrammatiko** [comment]
> No one force the developers to replace the sidebar with "search tools"

Well that was the reason @ciar4n didn't implement his mockup. Forcing people to use only the new way for filters is something that needs to be decided. I guess @wilsonge and @mbabker needs to give some feedback here

**Comment by mbabker** [comment]
We didn't 100% force Search Tools in part because Hathor didn't support it.  I think with 4.0 we can re-evaluate if we need to keep supporting two ways of handling search filters or sidebar navigation and cut the duplication if appropriate.

I will say though that these toolings very heavily favor if you're using the "classic" MVC stack in your extension.  If you're using your own RAD or the "no MVC" stack implementing these features isn't the easiest thing to do (I did pull it off in https://github.com/joomla-extensions/patchtester/blob/master/administrator/templates/atum/html/com_patchtester/pulls/default.php but it feels ugly to say the least).

File: administrator/components/com_languages/forms/filter_overrides.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+			label="JSEARCH_FILTER"
+			description="COM_LANGUAGES_SEARCH_IN_TITLE"
+			hint="JSEARCH_FILTER"
+		/>
+
+		<field
+				name="language_client"
+				type="list"
+				onchange="this.form.submit();"
+				default="en-GB0"
```
Comment (infograf768) [inline administrator/components/com_languages/forms/filter_overrides.xml]: en-GB

**Comment by Samgithub1997** [comment]
I have tested this item :red_circle: unsuccessfully on fcd5e88f454ddb962f66c96cd2cb3b842bc545b3<br /><br />When you click on categories or featured articles or fields or field groups , the UI is same as standard Joomla 4 (sidebar). The UI suggested by @ITPrism  (Navigation bar) is only functional for Articles. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19590">issues.joomla.org/tracker/joomla-cms/19590</a>.</sub>

**Comment by mbabker** [comment]
That's not an unsuccessful test.  This is a proof of concept PR and it did not change every component view.

**Comment by samarthsharma36** [comment]
I have tested this item :white_check_mark: successfully on fcd5e88f454ddb962f66c96cd2cb3b842bc545b3<br /><br />The UI suggested by @ITPrism  (Navigation bar) is only functional(successful) for Articles. The changes haven't been added for all component views.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19590">issues.joomla.org/tracker/joomla-cms/19590</a>.</sub>

**Comment by mbabker** [comment]
Still relevant given the new iteration on the template and the discussion about dropping the sidebar in full?

**Comment by wilsonge** [comment]
We're going to remove the submenu in entirety (although it's not been done yet) - also see https://github.com/joomla/40-backend-template/issues/454

**Comment by wilsonge** [comment]
Thankyou for the concept though :) very much appreciated

**Pull request closed without merging**